### PR TITLE
Make form pages more ergonomic

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Prereq to #650 

#### What's this PR do?
This adds utilities to avoid boilerplate around how we handle form events. You can look in the aforementioned PR to see how it's used. I did not add any usages of these in this PR to keep scope down.

#### How should this be manually tested?
Tests should pass

### CODE REVIEW

File: static/js/hoc/withForm.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        const { formBeginEdit } = this.props
+        formBeginEdit()
+      }
+
+      componentWillUnmount() {
+        const { formEndEdit } = this.props
+        formEndEdit()
+      }
+
+      onUpdate = (e: Object) => {
```
Comment (noisecapella) [inline static/js/hoc/withForm.js]: Can you add a test covering this function?

File: static/js/lib/redux_props_test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create()
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  describe("mergeAndInjectProps", () => {
+    it("should pass stateProps, dispatchProps, and ownProps into the passed method", () => {
```
Comment (noisecapella) [inline static/js/lib/redux_props_test.js]: Can you add some overlapping props verifying the order the spread operator gets applied?

**Comment by rhysyngsun** [comment]
Python tests are failing from an upstream commit in master, since this is just JS code, that shouldn't affect the ability to merge this.

**Comment by rhysyngsun** [comment]
@noisecapella tests are passing after rebase now

**Pull request merged**